src/app/zap-templates/templates/chip/helper.js

```diff
@@ -470,6 +470,27 @@ async function if_chip_enum(type, options)
   return templateUtil.templatePromise(this.global, result);
 }
 
+async function if_chip_complex(options)
+{
+  if (this.isArray) {
+    return options.fn(this);
+  }
+
+  if (this.type == 'array') {
```

Comment:
what is the difference between isArray and this.type == 'array' ? Please add a comment if this is some oddity regarding our data models. To me it looks like we have 2 ways of doing things, which is not ideal and should likely be cleaned up.

Reply:
https://stackoverflow.com/questions/22289727/difference-between-using-array-isarray-and-instanceof-array
Javascript has some oddities here, and it's pretty 'normal' to see these being mixed.

Reply:
I have added some comment to the code but I have also opened https://github.com/project-chip/zap/issues/412 to make ZAP consistent here.